code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductCatalogNotificationBuilder.kt

```diff
@@ -0,0 +1,335 @@
+        )
+        for (index in catalogItems.indices) {
+            val thumbImage = PushTemplateImageUtils.getCachedImage(catalogItems[index].img)
+
+            if (thumbImage == null) {
+                Log.trace(
+                    LOG_TAG,
+                    SELF_TAG,
+                    "No image found for catalog item thumbnail."
+                )
+            } else {
+                expandedLayout.setImageViewBitmap(thumbIds[index], thumbImage)
+            }
+
+            // create a pending intent for the thumbnail
+            val pendingIntentThumbnailInteraction = createThumbnailInteractionPendingIntent(
+                context,
+                broadcastReceiverClass,
+                channelIdToUse,
+                pushTemplate,
+                index
+            )
+            expandedLayout.setOnClickPendingIntent(
+                thumbIds[index],
+                pendingIntentThumbnailInteraction
+            )
+        }
+    }
+
+    /**
+     * Sets the CTA button for the product catalog notification.
+     *
+     * @param context the application [Context]
+     * @param trackerActivityClass the [Class] of the activity to set in the created pending intent for tracking purposes
+     * @param expandedLayout the [RemoteViews] object containing the expanded layout of the push template notification
+     * @param pushTemplate the [ProductCatalogPushTemplate] object containing the product catalog push template data
+     */
+    private fun setupCtaButton(
+        context: Context,
+        trackerActivityClass: Class<out Activity>?,
+        expandedLayout: RemoteViews,
+        pushTemplate: ProductCatalogPushTemplate
+    ) {
+        // apply the color to the cta button
+        expandedLayout.setCtaButtonColor(R.id.cta_button, pushTemplate.ctaButtonColor)
+
+        // apply text to the cta button
+        expandedLayout.setTextViewText(R.id.cta_button, pushTemplate.ctaButtonText)
+
+        // apply the open uri action to the cta button
+        expandedLayout.setOnClickPendingIntent(
+            R.id.cta_button,
+            PendingIntentUtils.createPendingIntent(
+                context,
+                trackerActivityClass,
+                pushTemplate.ctaButtonUri,
+                PushTemplateConstants.CatalogActionIds.CTA_BUTTON_CLICKED,
+                pushTemplate.tag,
+                pushTemplate.isNotificationSticky ?: false
+            )
+        )
+    }
+
+    /**
+     * Sets custom colors to the product catalog CTA button.
+     *
+     * @param containerViewId [Int] containing the resource id of the push template notification RemoteViews
+     * @param buttonColor [String] containing the hex color code for the cta button
+     */
+    private fun RemoteViews.setCtaButtonColor(
+        containerViewId: Int,
+        buttonColor: String?
+    ) {
+        // get custom color from hex string and set it the cta button
+        if (buttonColor.isNullOrEmpty()) {
```

Comment:
nit: this check is already done inside `setElementColor`, you can remove it from here

Reply:
i noticed the check within setElementColor was an empty check only while the check within `setCtaButtonColor` is for null or empty. i'll update this check to check for null as setElementColor expects the hex color string to be non null

Reply:
nit: Since this check is repeated again for `setCtaButtonTextColor`, might be easier to change `setElementColor` to accept optional string.

Reply:
I’ll make the change in setElementColor so we can get this cleaned up